Python/sysmodule.c

```diff
@@ -2521,6 +2517,10 @@ PyAPI_FUNC(int) PyUnstable_CopyPerfMapFile(const char* parent_filename) {
     }
     char buf[4096];
     PyThread_acquire_lock(perf_map_state.map_lock, 1);
+    FILE* from = fopen(parent_filename, "r");
```

Comment:
There's a behaviour change now: before we opened a file before the thread lock, now it is opened under a lock. What does this imply?

Reply:
Doesn't matter much although I would prefer to open it before acquiring lock.

Reply:
moved fopen to before PyThread_acquire_lock to satisfy both PR conversations.